### DISCUSSION TITLE
Wont recognize or download certain books

### DISCUSSION THREAD

**Discussion by SchnuSchnu** [audiamus/BookLibConnect#148] [Q&A]
Hi,
I connected to my audible account successfully. I wanted to convert my whole library and realized that some very few books wont even be listed in BookLibConnect. Why is that and how can I solve the problem?
Thanks

**Comment by audiamus** [comment]
The log files and the json responses from the server may reveal more information here. See chapter 4 in the manual how to find them and how to set the log level. The json files are usually in compact format and should be reformatted to make them legible. MS Visual Studio Code can do that.

**Reply by SchnuSchnu** [reply]
Thanks for your answer. However, I am a layman. I am not sure what you want me to do. Could you specify? Thank you for your understanding!

**Reply by audiamus** [reply]
BookLibConnect logs its activities in text form to a .log file. In addition, when it makes calls to the Audible server, it writes the server responses to files. These responses are also plain text, encoded in JSON format, hence written as .json files. Unfortunately, the original json responses are not directly human-readable, because they lack all spaces and line feeds. Therefore they need to be reformatted for the human eye, before we can inspect them. The log level, which you can set, see chapter 4, determines how detailed the log file will be and which server responses will be saved as temporary json files.   

And the log file and the associated json files may contain information, why certain books are not available.

However, before going into analysing the log and json files, another reason for missing books could be that Audible categorized those as "adult content" (their categorization is rather strange, sometimes). Adult products are not enabled by default in BookLibConnect, but there is a simple check-box in the main window to alter that.

**Reply by Valkenr** [reply]
I'm having a similar issue, one book isn't showing up. I tried toggling the adult filter and that didn't work..

Probing around the .json, with my very limited knowledge, I search for: 

title":"The Expanse"

and I see all the other books in the series, but book 9 is missing.

When I search for the "Leviathan Falls", the title of the book, all I get is stuff from product information sections where all the book titles are listed.

**Reply by audiamus** [reply]
If the book in question is not in the json library response or not listed there as a product, BookLibConnect will have no knowledge of that. 

Another reason for not finding it is a book purchased via a different account and your account has rights to view the library of the other account. BookLibConnect does not have access to such account cross links.

**Reply by Valkenr** [reply]
Apparently, I never bought it outright. Looks like it was added to the audible subscription catalogue after I started the series. Just had to go into the app and add it to my library, then it showed up.
